docs/INSTALL.md

```diff
@@ -11,27 +11,67 @@ In order to install the *job-executor-service* on the remote execution plane, so
 * `remoteControlPlane.api.token` - Keptn API Token (can be obtained from Bridge)
+* `remoteControlPlane.api.authMode` - Authentication mode which should be used when communicating with Keptn. Must be either `token` or `oauth`. Defaults to `token` if omitted
 
-**Example**
+**Example installation with token:**
 
 ```bash
+JES_VERSION=<VERSION> # e.g. 0.2.1
+
 KEPTN_API_PROTOCOL=http # or https
 KEPTN_API_HOST=<INSERT-YOUR-HOSTNAME-HERE> # e.g., 1.2.3.4.nip.io
  KEPTN_API_TOKEN=<INSERT-YOUR-KEPTN-API-TOKEN-HERE>
 
 TASK_SUBSCRIPTION=sh.keptn.event.remote-task.triggered
 
-helm upgrade --install --create-namespace -n <NAMESPACE> \
-  job-executor-service https://github.com/keptn-contrib/job-executor-service/releases/download/<VERSION>/job-executor-service-<VERSION>.tgz \
- --set remoteControlPlane.topicSubscription=${TASK_SUBSCRIPTION},remoteControlPlane.api.protocol=${KEPTN_API_PROTOCOL},remoteControlPlane.api.hostname=${KEPTN_API_HOST},remoteControlPlane.api.token=${KEPTN_API_TOKEN}
+helm upgrade --install --create-namespace -n <NAMESPACE> job-executor-service \
+    https://github.com/keptn-contrib/job-executor-service/releases/download/${JES_VERSION}/job-executor-service-${JES_VERSION}.tgz \
+    --set remoteControlPlane.topicSubscription=${TASK_SUBSCRIPTION} \
+    --set remoteControlPlane.api.protocol=${KEPTN_API_PROTOCOL} \
+    --set remoteControlPlane.api.hostname=${KEPTN_API_HOST} \
+    --set remoteControlPlane.api.token=${KEPTN_API_TOKEN}
 ```
 
 Please replace `<VERSION>` with the actual version you want to install from the compatibility matrix above or the
 [GitHub releases page](https://github.com/keptn-contrib/job-executor-service/releases).
 
 To verify that everything works you can visit Bridge, select a project, go to Uniform, and verify that `job-executor-service`  is registered as "remote execution plane" with the correct version and event type.
 
-**Example (with auto-detection)**
+**Example installation with OAuth:**
+
+If the oauth authentication mode should be used instead of the token, the following properties must be specified in addition to the setting `remoteControlPlane.api.authMode` to `oauth`:
+* `remoteControlPlane.api.oauth.clientId` - The OAuth client ID
+* `remoteControlPlane.api.oauth.clientSecret` - The OAuth client secret
+* `remoteControlPlane.api.oauth.clientDiscovery` - The well known discovery URL for the OAuth endpoint (e.g.: `https://your.sso.domain/.well-known/openid-configuration`)
+* `remoteControlPlane.api.oauth.scopes` - The scopes which should be requested by the job executor service
+
+```bash
+JES_VERSION=<VERSION> # e.g. 0.2.1
+
+KEPTN_API_PROTOCOL=http # or https
+KEPTN_API_HOST=<INSERT-YOUR-HOSTNAME-HERE> # e.g., 1.2.3.4.nip.io
+
+KEPTN_OAUTH_CLIENT_ID=<INSERT-YOUR-CLIENT_ID>
+ KEPTN_OAUTH_CLIENT_SECRET=<INSERT-YOUR-CLIENT_SECRET>
```

Comment:
OCD :stuck_out_tongue: 
```suggestion
KEPTN_OAUTH_CLIENT_SECRET=<INSERT-YOUR-CLIENT_SECRET>
```

Reply:
Bash doesn't save commands prefixed with a space into the history, if someone edits the instruction in a text editor and pastes them into the terminal the secret won't be saved there.

Reply:
Correct - please leave the space there for exactly that purpose ;)